sdks/java/extensions/ml/src/main/java/org/apache/beam/sdk/extensions/ml/RecommendationAIImportCatalogItems.java

```diff
@@ -0,0 +1,200 @@
+  public abstract @Nullable String catalogName();
+
+  /** @return Size of input elements batch to be sent in one request. */
+  public abstract Integer batchSize();
+
+  /**
+   * @return Time limit (in processing time) on how long an incomplete batch of elements is allowed
+   *     to be buffered.
+   */
+  public abstract Duration maxBufferingDuration();
+
+  public RecommendationAIImportCatalogItems withProjectId(String projectId) {
+    return this.toBuilder().setProjectId(projectId).build();
+  }
+
+  public RecommendationAIImportCatalogItems withCatalogName(String catalogName) {
+    return this.toBuilder().setCatalogName(catalogName).build();
+  }
+
+  public RecommendationAIImportCatalogItems withBatchSize(Integer batchSize) {
+    return this.toBuilder().setBatchSize(batchSize).build();
+  }
+
+  /**
+   * The transform converts the contents of input PCollection into {@link CatalogItem}s and then
+   * calls the Recommendation AI service to create the catalog item.
+   *
+   * @param input input PCollection
+   * @return PCollection after transformations
+   */
+  @Override
+  public PCollectionTuple expand(PCollection<KV<String, GenericJson>> input) {
+    return input
+        .apply(
+            "Batch Contents",
+            GroupIntoBatches.<String, GenericJson>ofSize(batchSize())
+                .withMaxBufferingDuration(maxBufferingDuration())
+                .withShardedKey())
+        .apply(
+            "Import CatalogItems",
+            ParDo.of(new ImportCatalogItems(projectId(), catalogName()))
+                .withOutputTags(SUCCESS_TAG, TupleTagList.of(FAILURE_TAG)));
+  }
+
+  @AutoValue.Builder
+  public abstract static class Builder {
+    /** @param projectId ID of Google Cloud project to be used for creating catalog items. */
+    public abstract Builder setProjectId(@Nullable String projectId);
+
+    /** @param catalogName Name of the catalog where the catalog items will be created. */
+    public abstract Builder setCatalogName(@Nullable String catalogName);
+
+    /**
+     * @param batchSize Amount of input elements to be sent to Recommendation AI service in one
+     *     request.
+     */
+    public abstract Builder setBatchSize(Integer batchSize);
+
+    /**
+     * @param maxBufferingDuration Time limit (in processing time) on how long an incomplete batch
+     *     of elements is allowed to be buffered.
+     */
+    public abstract Builder setMaxBufferingDuration(Duration maxBufferingDuration);
+
+    public abstract RecommendationAIImportCatalogItems build();
+  }
+
+  static class ImportCatalogItems
+      extends DoFn<KV<ShardedKey<String>, Iterable<GenericJson>>, CatalogItem> {
+    private final String projectId;
+    private final String catalogName;
+
+    /**
+     * @param projectId ID of GCP project to be used for creating catalog items.
+     * @param catalogName Catalog name for CatalogItem creation.
+     */
+    public ImportCatalogItems(String projectId, String catalogName) {
```

Comment:
you can remove `public` from this constructor so it can only be used within the package, and users wont accss it